src/core/python2js.c

```diff
@@ -34,12 +34,40 @@ typedef struct ConversionContext_s
                            JsRef key,
                            JsRef value);
   JsRef (*dict_postprocess)(struct ConversionContext_s context, JsRef dict);
+  JsRef jspostprocess_list;
   bool default_converter;
 } ConversionContext;
 
 JsRef
 _python2js(ConversionContext context, PyObject* x);
 
+EM_JS(void,
+      _python2js_addto_postprocess_list,
+      (JsRef idlist, JsRef idparent, JsRef idkey, PyObject* value),
+      {
+        const list = Hiwire.get_value(idlist);
+        const parent = Hiwire.get_value(idparent);
+        const key = Hiwire.get_value(idkey);
+        list.push([ parent, key, value ]);
+      });
+
+EM_JS(void, _python2js_handle_postprocess_list, (JsRef idlist, JsRef idcache), {
+  const list = Hiwire.get_value(idlist);
+  const cache = Hiwire.get_value(idcache);
+  for (const[parent, key, value] of list) {
+    let out_value = Hiwire.get_value(cache.get(value));
+    // clang-format off
+    if(parent.constructor.name === "Map"){
+      parent.set(key, out_value)
+    } else {
+      // This is unfortunately a bit of a hack, if user does something weird
+      // enough in dict_converter then it won't work.
+      parent[key] = out_value;
```

Comment:
It is very tricky to deal with various edge cases... but yeah we can handle them when it happens.